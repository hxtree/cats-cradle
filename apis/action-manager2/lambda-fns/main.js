"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const createNote_1 = require("./createNote");
const deleteNote_1 = require("./deleteNote");
const getNoteById_1 = require("./getNoteById");
const listNotes_1 = require("./listNotes");
const updateNote_1 = require("./updateNote");
exports.handler = async (event) => {
    switch (event.info.fieldName) {
        case "getNoteById":
            return await (0, getNoteById_1.default)(event.arguments.noteId);
        case "createNote":
            return await (0, createNote_1.default)(event.arguments.note);
        case "listNotes":
            return await (0, listNotes_1.default)();
        case "deleteNote":
            return await (0, deleteNote_1.default)(event.arguments.noteId);
        case "updateNote":
            return await (0, updateNote_1.default)(event.arguments.note);
        default:
            return null;
    }
};
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoibWFpbi5qcyIsInNvdXJjZVJvb3QiOiIiLCJzb3VyY2VzIjpbIm1haW4udHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6Ijs7QUFBQSw2Q0FBc0M7QUFDdEMsNkNBQXNDO0FBQ3RDLCtDQUF3QztBQUN4QywyQ0FBb0M7QUFDcEMsNkNBQXNDO0FBYXRDLE9BQU8sQ0FBQyxPQUFPLEdBQUcsS0FBSyxFQUFFLEtBQWtCLEVBQUUsRUFBRTtJQUMzQyxRQUFRLEtBQUssQ0FBQyxJQUFJLENBQUMsU0FBUyxFQUFFO1FBQzFCLEtBQUssYUFBYTtZQUNkLE9BQU8sTUFBTSxJQUFBLHFCQUFXLEVBQUMsS0FBSyxDQUFDLFNBQVMsQ0FBQyxNQUFNLENBQUMsQ0FBQztRQUNyRCxLQUFLLFlBQVk7WUFDYixPQUFPLE1BQU0sSUFBQSxvQkFBVSxFQUFDLEtBQUssQ0FBQyxTQUFTLENBQUMsSUFBSSxDQUFDLENBQUM7UUFDbEQsS0FBSyxXQUFXO1lBQ1osT0FBTyxNQUFNLElBQUEsbUJBQVMsR0FBRSxDQUFDO1FBQzdCLEtBQUssWUFBWTtZQUNiLE9BQU8sTUFBTSxJQUFBLG9CQUFVLEVBQUMsS0FBSyxDQUFDLFNBQVMsQ0FBQyxNQUFNLENBQUMsQ0FBQztRQUNwRCxLQUFLLFlBQVk7WUFDYixPQUFPLE1BQU0sSUFBQSxvQkFBVSxFQUFDLEtBQUssQ0FBQyxTQUFTLENBQUMsSUFBSSxDQUFDLENBQUM7UUFDbEQ7WUFDSSxPQUFPLElBQUksQ0FBQztLQUNuQjtBQUNMLENBQUMsQ0FBQSIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCBjcmVhdGVOb3RlIGZyb20gJy4vY3JlYXRlTm90ZSc7XHJcbmltcG9ydCBkZWxldGVOb3RlIGZyb20gJy4vZGVsZXRlTm90ZSc7XHJcbmltcG9ydCBnZXROb3RlQnlJZCBmcm9tICcuL2dldE5vdGVCeUlkJztcclxuaW1wb3J0IGxpc3ROb3RlcyBmcm9tICcuL2xpc3ROb3Rlcyc7XHJcbmltcG9ydCB1cGRhdGVOb3RlIGZyb20gJy4vdXBkYXRlTm90ZSc7XHJcbmltcG9ydCBOb3RlIGZyb20gJy4vTm90ZSc7XHJcblxyXG50eXBlIEFwcFN5bmNFdmVudCA9IHtcclxuICAgaW5mbzoge1xyXG4gICAgIGZpZWxkTmFtZTogc3RyaW5nXHJcbiAgfSxcclxuICAgYXJndW1lbnRzOiB7XHJcbiAgICAgbm90ZUlkOiBzdHJpbmcsXHJcbiAgICAgbm90ZTogTm90ZVxyXG4gIH1cclxufVxyXG5cclxuZXhwb3J0cy5oYW5kbGVyID0gYXN5bmMgKGV2ZW50OkFwcFN5bmNFdmVudCkgPT4ge1xyXG4gICAgc3dpdGNoIChldmVudC5pbmZvLmZpZWxkTmFtZSkge1xyXG4gICAgICAgIGNhc2UgXCJnZXROb3RlQnlJZFwiOlxyXG4gICAgICAgICAgICByZXR1cm4gYXdhaXQgZ2V0Tm90ZUJ5SWQoZXZlbnQuYXJndW1lbnRzLm5vdGVJZCk7XHJcbiAgICAgICAgY2FzZSBcImNyZWF0ZU5vdGVcIjpcclxuICAgICAgICAgICAgcmV0dXJuIGF3YWl0IGNyZWF0ZU5vdGUoZXZlbnQuYXJndW1lbnRzLm5vdGUpO1xyXG4gICAgICAgIGNhc2UgXCJsaXN0Tm90ZXNcIjpcclxuICAgICAgICAgICAgcmV0dXJuIGF3YWl0IGxpc3ROb3RlcygpO1xyXG4gICAgICAgIGNhc2UgXCJkZWxldGVOb3RlXCI6XHJcbiAgICAgICAgICAgIHJldHVybiBhd2FpdCBkZWxldGVOb3RlKGV2ZW50LmFyZ3VtZW50cy5ub3RlSWQpO1xyXG4gICAgICAgIGNhc2UgXCJ1cGRhdGVOb3RlXCI6XHJcbiAgICAgICAgICAgIHJldHVybiBhd2FpdCB1cGRhdGVOb3RlKGV2ZW50LmFyZ3VtZW50cy5ub3RlKTtcclxuICAgICAgICBkZWZhdWx0OlxyXG4gICAgICAgICAgICByZXR1cm4gbnVsbDtcclxuICAgIH1cclxufSJdfQ==