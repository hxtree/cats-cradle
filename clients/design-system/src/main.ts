export * from './components';
export * from './game-assets';
export * from './types';
export * from './utils';
