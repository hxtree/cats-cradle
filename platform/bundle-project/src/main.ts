import fs from 'fs';
import { Project } from '@pnpm/types';
import path from 'path';
import { argsParserPromise } from './util/args-parser.js';
import { Workspace } from './util/workspace.js';
import { PackageBundler } from './util/package-bundler.js';
import { Config } from './types/cli.js';
import { archive } from './util/archive';

export const main = async () => {
  console.log('start bundle project...');

  const dirname = path.dirname(__dirname);
  const workspaceRootDir = Workspace.getRootDir(dirname);

  const argsParser = await argsParserPromise;

  const targetPackageNameOrPath = argsParser._[0];

  console.log(targetPackageNameOrPath);

  if (!targetPackageNameOrPath || typeof targetPackageNameOrPath !== 'string') {
    throw Error('Target package name or path is required');
  }

  const config: Config = {
    outDir: `${workspaceRootDir}dist/${targetPackageNameOrPath.replace(
      '@cats-cradle/',
      '',
    )}`,
    overwrite: true,
    workspaceDependenciesFolder: 'workspace-dependencies',
  };

  console.log(`bundling ${targetPackageNameOrPath} to ${config.outDir}`);

  const workspacePackages = await Workspace.getPackages(workspaceRootDir);
  const workspace = new Workspace(workspaceRootDir, workspacePackages);

  const targetPackage: Project | null = workspace.tryToResolvePackage(
    targetPackageNameOrPath,
  );
  if (!targetPackage) {
    throw Error("Couldn't find target package inside workspace");
  }

  const packageBundler = new PackageBundler(workspace, targetPackage, config);

  packageBundler.createDistFolder();
  packageBundler.copyTargetPackageFiles();
  packageBundler.copyTargetPackageWorkspaceDependencies();

  console.log(`archive ${config.outDir} to ${config.outDir}.zip`);

  archive(config.outDir, `${config.outDir}.zip`);
};
