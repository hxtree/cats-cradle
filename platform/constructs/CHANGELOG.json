{
  "name": "@cats-cradle/constructs",
  "entries": [
    {
      "version": "0.0.13",
      "tag": "@cats-cradle/constructs_v0.0.13",
      "date": "Tue, 05 Mar 2024 05:29:22 GMT",
      "comments": {
        "patch": [
          {
            "comment": "update deps"
          }
        ],
        "dependency": [
          {
            "comment": "Updating dependency \"@cats-cradle/base-nodejs\" to `1.0.12`"
          }
        ]
      }
    },
    {
      "version": "0.0.12",
      "tag": "@cats-cradle/constructs_v0.0.12",
      "date": "Sun, 04 Feb 2024 05:29:40 GMT",
      "comments": {
        "dependency": [
          {
            "comment": "Updating dependency \"@cats-cradle/base-nodejs\" to `1.0.11`"
          }
        ]
      }
    },
    {
      "version": "0.0.11",
      "tag": "@cats-cradle/constructs_v0.0.11",
      "date": "Sat, 03 Feb 2024 08:29:12 GMT",
      "comments": {
        "patch": [
          {
            "comment": "api gateway construct update"
          }
        ]
      }
    },
    {
      "version": "0.0.10",
      "tag": "@cats-cradle/constructs_v0.0.10",
      "date": "Fri, 02 Feb 2024 05:50:03 GMT",
      "comments": {
        "patch": [
          {
            "comment": "add dns api endpoint"
          }
        ]
      }
    },
    {
      "version": "0.0.9",
      "tag": "@cats-cradle/constructs_v0.0.9",
      "date": "Thu, 28 Dec 2023 16:54:41 GMT",
      "comments": {
        "patch": [
          {
            "comment": "upgrade typescript 5.3.3"
          }
        ],
        "dependency": [
          {
            "comment": "Updating dependency \"@cats-cradle/base-nodejs\" to `1.0.10`"
          }
        ]
      }
    },
    {
      "version": "0.0.8",
      "tag": "@cats-cradle/constructs_v0.0.8",
      "date": "Tue, 12 Dec 2023 03:15:41 GMT",
      "comments": {
        "dependency": [
          {
            "comment": "Updating dependency \"@cats-cradle/base-nodejs\" to `1.0.9`"
          }
        ]
      }
    },
    {
      "version": "0.0.7",
      "tag": "@cats-cradle/constructs_v0.0.7",
      "date": "Mon, 13 Nov 2023 02:11:53 GMT",
      "comments": {
        "patch": [
          {
            "comment": "update packages"
          }
        ]
      }
    },
    {
      "version": "0.0.6",
      "tag": "@cats-cradle/constructs_v0.0.6",
      "date": "Sun, 05 Nov 2023 18:41:09 GMT",
      "comments": {
        "dependency": [
          {
            "comment": "Updating dependency \"@cats-cradle/base-nodejs\" to `1.0.8`"
          }
        ]
      }
    },
    {
      "version": "0.0.5",
      "tag": "@cats-cradle/constructs_v0.0.5",
      "date": "Sat, 04 Nov 2023 19:40:56 GMT",
      "comments": {
        "patch": [
          {
            "comment": "unify unit tests"
          }
        ],
        "dependency": [
          {
            "comment": "Updating dependency \"@cats-cradle/base-nodejs\" to `1.0.7`"
          }
        ]
      }
    },
    {
      "version": "0.0.4",
      "tag": "@cats-cradle/constructs_v0.0.4",
      "date": "Sat, 14 Oct 2023 22:49:51 GMT",
      "comments": {
        "patch": [
          {
            "comment": "add layers prop"
          }
        ]
      }
    },
    {
      "version": "0.0.3",
      "tag": "@cats-cradle/constructs_v0.0.3",
      "date": "Wed, 04 Oct 2023 03:44:49 GMT",
      "comments": {
        "patch": [
          {
            "comment": "bump packages"
          }
        ]
      }
    },
    {
      "version": "0.0.2",
      "tag": "@cats-cradle/constructs_v0.0.2",
      "date": "Tue, 27 Jun 2023 18:44:40 GMT",
      "comments": {
        "patch": [
          {
            "comment": "upgrade to node 18 lts"
          }
        ]
      }
    }
  ]
}
