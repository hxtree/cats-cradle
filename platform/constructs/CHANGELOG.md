# Change Log - @cats-cradle/constructs

This log was last generated on Sun, 04 Feb 2024 05:29:40 GMT and should not be manually modified.

## 0.0.12
Sun, 04 Feb 2024 05:29:40 GMT

_Version update only_

## 0.0.11
Sat, 03 Feb 2024 08:29:12 GMT

### Patches

- api gateway construct update

## 0.0.10
Fri, 02 Feb 2024 05:50:03 GMT

### Patches

- add dns api endpoint

## 0.0.9
Thu, 28 Dec 2023 16:54:41 GMT

### Patches

- upgrade typescript 5.3.3

## 0.0.8
Tue, 12 Dec 2023 03:15:41 GMT

_Version update only_

## 0.0.7
Mon, 13 Nov 2023 02:11:53 GMT

### Patches

- update packages

## 0.0.6
Sun, 05 Nov 2023 18:41:09 GMT

_Version update only_

## 0.0.5
Sat, 04 Nov 2023 19:40:56 GMT

### Patches

- unify unit tests

## 0.0.4
Sat, 14 Oct 2023 22:49:51 GMT

### Patches

- add layers prop

## 0.0.3
Wed, 04 Oct 2023 03:44:49 GMT

### Patches

- bump packages

## 0.0.2
Tue, 27 Jun 2023 18:44:40 GMT

### Patches

- upgrade to node 18 lts

