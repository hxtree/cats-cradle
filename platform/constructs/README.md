L1 constructs
L2 constructs

https://docs.aws.amazon.com/cdk/v2/guide/constructs.html