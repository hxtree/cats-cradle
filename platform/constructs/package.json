{
  "name": "@cats-cradle/constructs",
  "version": "0.0.1",
  "module": "commonjs",
  "main": "./dist/src/index.js",
  "scripts": {
    "build": "tsc",
    "watch": "tsc -w",
    "test": "jest",
    "lint": "eslint --format visualstudio \"./{stacks,src}/*.ts\"  --fix",
    "lint:ci": "eslint --format visualstudio \"./{stacks,src}/*.ts\"  --fix-dry-run",
    "depcheck": "npx depcheck"
  },
  "dependencies": {
    "aws-lambda": "1.0.7",
    "class-transformer": "0.5.1",
    "class-validator": "0.13.2",
    "@nestjs/common": "9.4.0",
    "@nestjs/core": "9.2.1",
    "@nestjs/terminus": "9.1.1",
    "@nestjs/platform-express": "9.3.9",
    "aws-serverless-express": "~3.4.0",
    "@vendia/serverless-express": "4.10.1",
    "express": "4.18.2",
    "node-cache": "5.1.2",
    "reflect-metadata": "0.1.13",
    "rxjs": "7.5.6",
    "mongoose": "~6.9.1",
    "source-map-support": "~0.5.21",
    "@nestjs/swagger": "~6.1.3",
    "uuid": "~9.0.0",
    "lodash": "~4.17.21",
    "@types/lodash": "~4.14.191",
    "@cats-cradle/bundlers": "workspace:*"
  },
  "devDependencies": {
    "aws-cdk-lib": "2.47.0",
    "constructs": "^10.1.303",
    "@types/jest": "28.1.8",
    "@types/node": "^16.0.0",
    "@types/aws-lambda": "~8.10.114",
    "jest": "29.5.0",
    "ts-jest": "29.1.0",
    "typescript": "5.0.2",
    "ts-node": "10.9.1",
    "esbuild": "~0.17.5",
    "fs": "~0.0.1-security",
    "path": "~0.12.7",
    "@types/express": "~4.17.14",
    "@types/uuid": "~8.3.4",
    "eslint": "8.38.0",
    "@cats-cradle/eslint-config": "1.0.5"
  },
  "exports": {
    ".": "./dist/src/index.js",
    "./api-endpoint": "./dist/src/api-endpoint/index.js",
    "./aws-account": "./dist/src/aws-accounts/index.js",
    "./microservice": "./dist/src/microservice/index.js",
    "./nestjs": "./dist/src/nestjs/index.js"
  }
}
