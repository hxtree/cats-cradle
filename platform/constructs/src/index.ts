export * from './microservice/microservice';
export * from './api-endpoint/api-endpoint';
export * from './api-endpoint/get-base-url';
export * from './aws-accounts';
export * from './nestjs/nestjs';
