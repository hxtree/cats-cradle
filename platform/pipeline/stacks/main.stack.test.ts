import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { awsAccounts } from '@cats-cradle/constructs';
import { MainStack } from './main.stack';

describe('MainStack', () => {
  it('should match snapshot test', () => {
    const app = new cdk.App();

    const stack = new MainStack(app, 'MainStack', {
      env: {
        account: awsAccounts.tools.accountId,
        region: awsAccounts.tools.region,
      },
    });
    const template = Template.fromStack(stack);

    expect(template).toMatchSnapshot();
  });
});
