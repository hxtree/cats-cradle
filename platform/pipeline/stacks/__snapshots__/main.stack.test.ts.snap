// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainStack should match snapshot test 1`] = `
{
  "Outputs": {
    "AWSACCESSKEY": {
      "Value": {
        "Fn::GetAtt": [
          "GithubUploadStackNestedStackGithubUploadStackNestedStackResourceE585A496",
          "Outputs.MainStackGithubUploadStackGithubUploadStackgithubciaccesskey13F50D46Ref",
        ],
      },
    },
    "AWSBUCKET": {
      "Value": {
        "Fn::GetAtt": [
          "GithubUploadStackNestedStackGithubUploadStackNestedStackResourceE585A496",
          "Outputs.MainStackGithubUploadStackGithubUploadStackrushdeploybucket9E28709DRef",
        ],
      },
    },
    "AWSREGION": {
      "Value": "us-east-2",
    },
    "AWSSECRETACCESSKEY": {
      "Value": {
        "Fn::GetAtt": [
          "GithubUploadStackNestedStackGithubUploadStackNestedStackResourceE585A496",
          "Outputs.MainStackGithubUploadStackGithubUploadStackgithubciaccesskey13F50D46SecretAccessKey",
        ],
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "GithubUploadStackNestedStackGithubUploadStackNestedStackResourceE585A496": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "TemplateURL": {
          "Fn::Join": [
            "",
            [
              "https://s3.us-east-2.",
              {
                "Ref": "AWS::URLSuffix",
              },
              "/cdk-hnb659fds-assets-760440398296-us-east-2/a70ffce7800b738ca6ad97b052845db25a54448d3678ab89ca8434bbb218db2f.json",
            ],
          ],
        },
      },
      "Type": "AWS::CloudFormation::Stack",
      "UpdateReplacePolicy": "Delete",
    },
    "LuckByDicePipelineStackNestedStackLuckByDicePipelineStackNestedStackResourceEBE4B2DD": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "Parameters": {
          "referencetoMainStackGithubUploadStackNestedStackGithubUploadStackNestedStackResource93F31631OutputsMainStackGithubUploadStackGithubUploadStackrushdeploybucket9E28709DArn": {
            "Fn::GetAtt": [
              "GithubUploadStackNestedStackGithubUploadStackNestedStackResourceE585A496",
              "Outputs.MainStackGithubUploadStackGithubUploadStackrushdeploybucket9E28709DArn",
            ],
          },
          "referencetoMainStackGithubUploadStackNestedStackGithubUploadStackNestedStackResource93F31631OutputsMainStackGithubUploadStackGithubUploadStackrushdeploybucket9E28709DRef": {
            "Fn::GetAtt": [
              "GithubUploadStackNestedStackGithubUploadStackNestedStackResourceE585A496",
              "Outputs.MainStackGithubUploadStackGithubUploadStackrushdeploybucket9E28709DRef",
            ],
          },
        },
        "TemplateURL": {
          "Fn::Join": [
            "",
            [
              "https://s3.us-east-2.",
              {
                "Ref": "AWS::URLSuffix",
              },
              "/cdk-hnb659fds-assets-760440398296-us-east-2/f5ea1b9424811e68e4119da217da6bc07b865eba84b2083133f7474b5720c03a.json",
            ],
          ],
        },
      },
      "Type": "AWS::CloudFormation::Stack",
      "UpdateReplacePolicy": "Delete",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
